kcl/src/main/scala/kinesis4cats/kcl/KCLConsumer.scala

```diff
@@ -175,6 +178,84 @@ object KCLConsumer {
     )(cb)
     .map(new KCLConsumer[F](_))
 
+  def builder[F[_]: Async](
+      appName: String,
+      streamTracker: StreamTracker
+  ): KCLConsumerBuilder[F] =
+    new KCLConsumerBuilder[F](
+      appName = appName,
+      streamTracker = streamTracker,
+      mkWorkerId = UUIDGen[F].randomUUID.map(_.toString),
+      kinesisClientResource = Resource
+        .fromAutoCloseable(
+          Sync[F].delay(KinesisAsyncClient.builder().build())
+        ),
+      dynamoClientResource = Resource.fromAutoCloseable(
+        Sync[F].delay(DynamoDbAsyncClient.builder().build())
+      ),
+      cloudWatchClientResource = Resource.fromAutoCloseable(
+        Sync[F].delay(CloudWatchAsyncClient.builder().build())
+      ),
+      callback = Function.const(Applicative[F].unit),
+      config = identity
+    )
+
+  final case class KCLConsumerBuilder[F[_]: Async] private[KCLConsumer] (
+      private val appName: String,
+      private val streamTracker: StreamTracker,
+      private val mkWorkerId: F[String],
+      private val kinesisClientResource: Resource[F, KinesisAsyncClient],
+      private val dynamoClientResource: Resource[F, DynamoDbAsyncClient],
+      private val cloudWatchClientResource: Resource[F, CloudWatchAsyncClient],
+      private val callback: List[CommittableRecord[F]] => F[Unit],
```

Comment:
I feel like this should be part of the "bare minimum" for a KCL consumer, but I could be convinced to leave it as defaulted.

Reply:
right. my only problem with that is that a callback like this deserves a separate argument list... and I don't like these so much 💀

but it's probably a lesser evil. People are bound to just forget it and spend an hour debugging why their consumer isn't working because they forgot to actually pass a callback.